mavutil.py

```diff
@@ -97,7 +97,7 @@ def add_message(messages, mtype, msg):
         messages[mtype] = msg
         return
     instance_value = getattr(msg, msg._instance_field)
-    if not mtype in messages:
+    if mtype not in messages or messages[mtype]._instances is None:
```

Comment:
Why not check for `msg.get_type() == 'BAD_DATA'` at the top of this method, and not do silly things with invalid data?

Reply:
Thanks for the review! @peterbarker

That fixes the bug, and pytest succeeds. But has the side effect of never adding BAD_DATA messages to mavfile sysid_state.messages, which is a slight change to behavior. I looked for all usages in this repo, and I don't see any problems, and in fact I imagine it might fix an obscure class of bugs. If you are OK w/ this I'll make the change.

Reply:
I made this change and rebased. Tests pass locally. Thanks.